hydrator-common/src/main/java/co/cask/hydrator/common/SourceInputFormatProvider.java

```diff
@@ -38,6 +40,12 @@ public SourceInputFormatProvider(Class<? extends InputFormat> inputFormatClass,
 
   public SourceInputFormatProvider(String inputFormatClassName, Configuration hConf) {
     Map<String, String> config = new HashMap<>();
+    // Unset the values of the default filesystem and implementation of the filesystem
```

Comment:
However this would mean changing the `SourceInputFormatProvider` constructor to accept the `Job` instance and this also seems like an additional layer to get the configuration as
`JobUtils.getConfiguration(job)`.
Don't have strong opinion on this. Let me know your thoughts.